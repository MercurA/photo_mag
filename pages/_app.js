import '../styles/globals.css'
import { createContext, useReducer } from "react"
import reducer, { initialState } from "../utils/state/reducer"

export const AppContext = createContext(null)

const MyApp = ({ Component, pageProps }) => {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <AppContext.Provider value={{ ...state, dispatch }}>
      <Component {...pageProps} />
    </AppContext.Provider>
  )
}

export default MyApp