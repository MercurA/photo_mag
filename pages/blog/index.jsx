import DesktopBlogContainer from '../../containers/DesktopBlogContainer'

const Blog = () => {
    return (
        <DesktopBlogContainer />
    )
}

export default Blog;