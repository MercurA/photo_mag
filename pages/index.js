import { createContext, useReducer } from "react";
import CardContainer from "../components/card_image/CardContainer";
import Menu from "../components/menu/Menu";
import MusicContainer from "../components/music/MusicContainer";
import reducer, { initialState } from "../utils/state/reducer";

export const AppContext = createContext(null)

export default function Home() {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <AppContext.Provider value={state}>
      <div >
        <Menu dispatch={dispatch}/>
        <CardContainer />
        <MusicContainer dispatch={dispatch}/>
      </div>
    </AppContext.Provider>
  )
}
