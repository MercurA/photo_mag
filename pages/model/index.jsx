import { useRef, useState } from 'react'
import styles from './model.module.css'
import { Canvas, useFrame } from '@react-three/fiber'

const Sphere = ({position, size, color}) => {
    const [hovered, setHovered] = useState(false)
    const sphereRef = useRef()
    useFrame((state, delta) => {
        sphereRef.current.rotation.y += delta/6
    })
    return (
        <mesh ref={sphereRef} onPointerOver={(e) => {e.stopPropagation(); setHovered(!hovered)}}>
            <sphereGeometry />
            <meshStandardMaterial color={hovered ? color : 'lightblue'} wireframe/>
        </mesh>
    )
}

const Model = () => {
    return (
        <div className={styles.container}>
            <Canvas >
                <directionalLight position={[0, 0, 7]} />
                <Sphere position={[0,0,0]} size={[1,1,1]} color={'orange'}/>
            </Canvas>
        </div>
    )
}

export default Model