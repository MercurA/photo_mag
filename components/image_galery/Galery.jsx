import { useContext, useEffect, useRef, useState } from 'react'
import styles from './galery.module.css'
import { AppContext } from '../../pages'
import Image from 'next/image'
import { ACTIONS } from '../../utils/state/reducer'

const Galery = ({ dispatch }) => {
    const context = useContext(AppContext)
    const { imageList } = context
    const scrollContainerRef = useRef(null)
    const itemRef = useRef(null)
    const [items, setItems] = useState(imageList)

    useEffect(() => {
        const current = scrollContainerRef.current
        current.addEventListener('wheel', handleScroll)

        return () => current.removeEventListener('wheel', handleScroll)
    },[items])

    const handleScroll = (e) => {
        if (e.deltaY > 0) {
            setItems((prevItems) => {
                dispatch({
                    type: ACTIONS.setCurrentImageToDisplay,
                    payload: prevItems[1]
                })
                return [
                    ...prevItems.slice(1),
                    prevItems[0]
                ]
            });
        } else if (e.deltaY < 0) {
            setItems((prevItems) => {
                dispatch({
                    type: ACTIONS.setCurrentImageToDisplay,
                    payload: prevItems[prevItems.length - 1]
                })
                return [
                    prevItems[prevItems.length - 1],
                    ...prevItems.slice(0, -1),
                ]
            });
        }
    }

    return (
        <div
            ref={scrollContainerRef}
            className={styles.container}
        >
            {items.map((image, index) => (
                <div className={styles.item} key={index} ref={itemRef}>
                    <Image
                        src={image}
                        alt={''}
                        fill={true}
                        className='image'
                    />
                </div>
            ))}
        </div>
    )
}

export default Galery