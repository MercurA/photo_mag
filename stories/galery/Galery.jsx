import { useContext, useEffect, useRef, useState } from 'react'
import Image from 'next/image'
import './galery.css'
import { AppContext } from '../../pages'

import {data} from './data'

export const Galery = () => {
    const context = data

    const { imageList, currentImageCollection, dispatch } = context
    
    const scrollContainerRef = useRef(null)
    const itemRef = useRef(null)
    const [items, setItems] = useState(imageList[currentImageCollection])

    useEffect(() => {
        setItems(imageList[currentImageCollection])
    },[currentImageCollection, imageList])


    useEffect(() => {
        const current = scrollContainerRef?.current

        const handleScroll = (e) => {
            if (e.deltaY > 0) {
                setItems((prevItems) => {
                    return [
                        ...prevItems.slice(1),
                        prevItems[0]
                    ]
                });
            } else if (e.deltaY < 0) {
                setItems((prevItems) => {
                    return [
                        prevItems[prevItems.length - 1],
                        ...prevItems.slice(0, -1),
                    ]
                });
            }
        }
        current.addEventListener('touchstart', handleScroll)
        current.addEventListener('wheel', handleScroll)

        return () => {
            current.removeEventListener('touchstart', handleScroll)
            current.removeEventListener('wheel', handleScroll)
        }
    }, [items, dispatch, scrollContainerRef])


    return (
        <>  
            <div
                ref={scrollContainerRef}
                className={'container'}
                id="galery"
            >   
                <div className={'focus'}></div>
                {items?.map((image, index) => (
                    <div className={'item'} key={index} ref={itemRef}>
                        <Image
                            src={image.path}
                            alt={''}
                            fill={true}
                            className='image'
                        />
                    </div>
                ))}
            </div>
            <div></div>
        </>
    )   
}

Galery.propType = {

}


