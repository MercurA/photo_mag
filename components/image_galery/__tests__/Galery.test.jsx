import { render } from "@testing-library/react"
import { AppContext } from "../../../pages"
import Galery from "../Galery"

describe('Galery', () => {
    const state = {
        imageList: ['/images/1.jpg']
    }
    it('should render correctly', () => {
        const comp = render(
            <AppContext.Provider value={state}>
                <Galery imageList={state.imageList}/>
            </AppContext.Provider>
        )

        expect(comp).toMatchSnapshot()
    })
})