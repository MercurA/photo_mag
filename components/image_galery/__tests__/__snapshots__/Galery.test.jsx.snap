// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Galery should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="item"
        >
          <img
            alt=""
            class="image"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2F1.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2F1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2F1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2F1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2F1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2F1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2F1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2F1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2F1.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="item"
      >
        <img
          alt=""
          class="image"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimages%2F1.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2F1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2F1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2F1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2F1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2F1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2F1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2F1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2F1.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
